lib/decompress_impl.ml

```diff
@@ -1538,6 +1555,350 @@ module Zlib_deflate = struct
+    | RFC1951_deflate.Error (d, exn) -> error {t with d} (RFC1951 exn)
+
+  let nop k src dst t = k src dst t
+
+  let fextra extra k src dst t =
+    let extra_l = String.length extra in
+    ( KHeader.put_short_lsb extra_l
+    @@ KHeader.put_string extra
+    @@ fun src dst t -> k src dst t )
+      src dst t
+
+  let fname name k src dst t =
+    ( KHeader.put_string name
+    @@ KHeader.put_byte 0
+    @@ fun src dst t -> k src dst t )
+      src dst t
+
+  let fcomment comment k src dst t =
+    ( KHeader.put_string comment
+    @@ KHeader.put_byte 0
+    @@ fun src dst t -> k src dst t )
+      src dst t
+
+  let fcrc16 crc16 k src dst t =
+    ( KHeader.put_short_lsb (Optint.to_int crc16)
+    @@ fun src dst t -> k src dst t )
+      src dst t
+
+  let get_option k = function Some opt -> k opt | None -> nop
+
+  let header src dst t =
+    let id1 = 31 in
+    let id2 = 139 in
+    let cm = 8 in
+    let flg = if t.text then 0xb1 else 0 in
+    let flg = if t.crc16 <> None then flg lor 0b10 else flg in
+    let flg = if t.extra <> None then flg lor 0b100 else flg in
+    let flg = if t.name <> None then flg lor 0b1000 else flg in
+    let flg = if t.comment <> None then flg lor 0b10000 else flg in
+    let mt0 = t.mtime land 0xFF in
+    let mt1 = (t.mtime lsr 8) land 0xFF in
+    let mt2 = (t.mtime lsr 16) land 0xFF in
+    let mt3 = t.mtime lsr 24 in
+    let xfl = 0 in
+    let os = t.os in
+    let fextra = get_option fextra t.extra in
+    let fname = get_option fname t.name in
+    let fcomment = get_option fcomment t.comment in
+    ( KHeader.put_byte id1
+    @@ KHeader.put_byte id2
+    @@ KHeader.put_byte cm
+    @@ KHeader.put_byte flg
+    @@ KHeader.put_byte mt0
+    @@ KHeader.put_byte mt1
+    @@ KHeader.put_byte mt2
+    @@ KHeader.put_byte mt3
+    @@ KHeader.put_byte xfl
+    @@ KHeader.put_byte os
+    @@ fextra
+    @@ fname
+    @@ fcomment
+    @@ fun src dst t ->
+    let fcrc16 = get_option fcrc16 t.crc16 in
+    let final _src _dst t =
+      Cont
+        { t with
+          d= {t.d with RFC1951_deflate.hold= 0; RFC1951_deflate.bits= 0}
+        ; z= Deflate }
+    in
+    (fcrc16 @@ final) src dst t )
+      src dst t
+
+  let eval src dst t =
+    let safe_src = Safe.ro t.d.RFC1951_deflate.wi src in
+    let safe_dst = Safe.wo t.d.RFC1951_deflate.wo dst in
+    let eval0 t =
+      match t.z with
+      | Header k -> k safe_src safe_dst t
+      | Deflate -> deflate safe_src safe_dst t
+      | Crc32 k -> k safe_src safe_dst t
+      | Size k -> k safe_src safe_dst t
+      | Finish -> ok t
+      | Exception exn -> error t exn
+    in
+    let rec loop t =
+      match eval0 t with
+      | Cont t -> loop t
+      | Wait t -> `Await t
+      | Flush t -> `Flush t
+      | Ok t -> `End t
+      | Error (t, exn) -> `Error (t, exn)
+    in
+    loop t
+
+  let default ~witness ?(text = false) ?(header_crc = false) ?(extra = None)
+      ?(name = None) ?(comment = None) ?(mtime = 0) ?(os = 255) level =
```

Comment:
When you write something like `?(extra = None)`, you can just write `?extra`. OCaml put the `None` value automatically.